sdk/rust/nym-sdk/examples/socks5.rs

```diff
@@ -0,0 +1,51 @@
+use nym_sdk::mixnet;
+
+#[tokio::main]
+async fn main() {
+    nym_bin_common::logging::setup_logging();
+
+    println!("Connecting receiver");
+    let mut receiving_client = mixnet::MixnetClient::connect_new().await.unwrap();
+
+    let config = mixnet::Config {
+        socks5_service_provider: Some(receiving_client.nym_address().to_string()),
+        ..Default::default()
+    };
+    let sending_client = mixnet::MixnetClientBuilder::new()
+        .config(config)
+        .build::<mixnet::EmptyReplyStorage>()
+        .await
+        .unwrap();
+
+    println!("Connecting sender");
+    let mut sending_client = sending_client.connect_to_mixnet().await.unwrap();
+    // wait until socks5 server is started
+    println!("Wait 5 seconds for the socks5 setup to be done");
+    std::thread::sleep(std::time::Duration::from_secs(5));
+
+    let proxy = reqwest::Proxy::all(format!(
+        "socks5h://127.0.0.1:{}",
+        nym_network_defaults::DEFAULT_SOCKS5_LISTENING_PORT
```

Comment:
Would be neat if you could get the port from querying the sending client instance

Reply:
I will do that as part of https://github.com/nymtech/nym/issues/3237 , and possibly expose other socks5-specific structures.